store/store.go

```diff
@@ -1177,6 +1180,18 @@ func (s *Store) Query(qr *proto.QueryRequest) ([]*proto.QueryRows, error) {
 	return s.db.Query(qr.Request, qr.Timings)
 }
 
+func (s *Store) leaderRead(qr *proto.QueryRequest) ([]*proto.QueryRows, error) {
+	future := s.raft.VerifyLeader()
```

Comment:
How does `VerifyLeader` actually work? I reviewed the Hashicorp docs, they simply state "ensure this peer is still the leader". How does that work? The Hashicorp code

From looking at the Hashicorp code the function claims to "ensure this peer is still the leader". What prevents the node losing leadership immediately after this call returns, but before it does the read?

From looking at the code it appears to tell the Followers to heartbeat immediately. Presumably that means that the Leader won't step down for at least an election timeout. Is that it?

Reply:
yes